### ISSUE TITLE
Exception in archive.extracted (File or directory does not exist)

### ISSUE THREAD

**Issue by mkotsbak** [saltstack/salt#47831]
### Description of Issue/Question

Exception I get:
```
    Function: archive.extracted
        Name: /opt
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1878, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1823, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/archive.py", line 1429, in extracted
                  test=__opts__['test'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1823, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/file.py", line 3098, in directory
                  full, ret, user, group, file_mode, None, follow_symlinks)
                File "/usr/lib/python2.7/site-packages/salt/modules/file.py", line 4397, in check_perms
                  perms['lattrs'] = ''.join(lsattr(name).get('name', ''))
                File "/usr/lib/python2.7/site-packages/salt/modules/file.py", line 552, in lsattr
                  raise SaltInvocationError("File or directory does not exist.")
              SaltInvocationError: File or directory does not exist.
```
### Setup
```
profile.foo_extract:
  archive.extracted:
    - name: /opt
    - source: /home/bar/foo_installed.tar.bz2
    - source_hash: 8aaa6b4587c5d6b27edea9d4d2e3f59da80f306660dff7d6dd795ca47ea1eea9
    - user: bar
    - group: bar
    - require:
        - cmd: profile.foo_download
```
### Steps to Reproduce Issue
Run salt-call state.highstate

### Versions Report
Salt Version:
           Salt: 2017.7.5
 
Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15rc1 (default, Apr 15 2018, 21:51:34)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-20-generic
         system: Linux
        version: Ubuntu 18.04 bionic

**Comment by gtmanfred** [comment]
I am unable to replicate this.

```
root@salt:/srv/salt# cat test.sls
bash:
  archive.extracted:
    - name: /opt
    - source: https://ftp.gnu.org/gnu/bash/bash-4.4.tar.gz
    - source_hash: d86b3392c1202e8ff5a423b302e6284db7f8f435ea9f39b5b1b20fd3ac36dfcb
    - user: kitchen
    - group: kitchen
root@salt:/srv/salt# salt-call --local state.apply test
local:
----------
          ID: bash
    Function: archive.extracted
        Name: /opt
      Result: True
     Comment: https://ftp.gnu.org/gnu/bash/bash-4.4.tar.gz extracted to /opt/, due to absence of one or more files/dirs
     Started: 13:16:25.297113
    Duration: 2549.655 ms
     Changes:
              ----------
              directories_created:
                  - /opt/
              extracted_files:
                  - bash-4.4
                  - bash-4.4/CWRU
                  - bash-4.4/builtins
                  - bash-4.4/cross-build
                  - bash-4.4/doc
                  - bash-4.4/examples
                  - bash-4.4/include
                  <snip>
              updated ownership:
                  True

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:   2.550 s
root@salt:/srv/salt# salt-call --version
salt-call 2017.7.5 (Nitrogen)
```

Can you provide me with some more information so that I can replicate the issue?

**Comment by mkotsbak** [comment]
I can't reproduce it with bash even with local downloaded archive. But I see that my archive has one soft link pointing to a non existing directory which might cause this.

**Comment by gtmanfred** [comment]
That could do it.

**Comment by mkotsbak** [comment]
Yes, I fixed the symlink to point to the relative file that is there. Then it works. Probably Salt is trying to follow the symlink which is indeed ponting to a file not found.

**Comment by gtmanfred** [comment]
Cool, i am going to mark this as a bug, but not a high priority.

Thanks for reporting and helping track it down.
Daniel

**Comment by xenophonf** [comment]
I hit this error using Salt 2018.3.2 on CentOS 7.5.1804 with the following file.directory state:

https://github.com/irtnog/comanage-formula/blob/44d628d485d22a4b99efab3b16f7c4494f1471c6/comanage/registry/init.sls#L63

There's a symlink at /opt/comanage-registry/app/webroot/img/logo.png, pointing to an optional logo file at ../../../local/webroot/img/logo.png.  This file doesn't exist on my system by design.

The `file.lsattr` execution module function should not return an invocation error in this case.  Perhaps it should return `None`, or maybe it shouldn't dereference the symlink.  In any case, this change broke production deployments of this app once we upgraded to 2018.3.0.  I've worked around it by hot-patching this function to return `None` instead of raising an exception.
